ggshield/core/scan/commit.py

```diff
@@ -73,6 +75,57 @@ def parser(commit: "Commit") -> Iterable[Scannable]:
 
         return Commit(sha, parser, info)
 
+    @staticmethod
+    def from_merge(
+        exclusion_regexes: Optional[Set[Pattern[str]]] = None,
+        merge_branch: str = "MERGE_HEAD",
+        cwd: Optional[Path] = None,
+    ) -> "Commit":
```

Comment:
Overall this function is a bit difficult to read because it mixes low-level aspects (parsing git command output) with high-level aspects (deciding which files to scan).

I made a few naming suggestions but the real readability improvement would probably be to turn `get_file_sha_from_ls_tree()` into a standalone function in this module and to create similar helper functions for the other low-level aspects, so that `from_merge()` only contains the high-level code.

Reply:
Should be better now :)

Reply:
Much nicer! ✨